src/hooks/useTimeout/index.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useEffect, useState } from 'react';
+
+const useTimeout = () => {
+  const [timerIds, setTimerIds] = useState([]);
+
+  // clear the timeout.
+  useEffect(() => {
+    return () => timerIds.map((timerId) => clearTimeout(timerId));
+  }, [timerIds]);
```

Comment:
Is it fine using `timerIds` as dependency? will it not clear existing timeouts each time you add a new one?

Reply:
But the test shows me that it is working fine, what I am missing?

Reply:
I think you're right. It causes me doubt, I will investigate